File: RestaurantBookingSystem_Backend/Model/Restaurant/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RestaurantBookingSystem.Model.Restaurant
{
    public class Payment
    {
        [Key]
        public int PaymentId { get; set; }

        [Required]
        [ForeignKey(nameof(Order))]
        public int OrderId { get; set; }

        [Required]
        [Range(0, 99999999.99)]
        public decimal Amount { get; set; }

        [Required]
        public DateTime PaymentDate { get; set; } = DateTime.Now;

        [Required]
        public PayMethod PayMethod { get; set; } = PayMethod.UPI;

        [Required]
        public PaymentStatus Status { get; set; } = PaymentStatus.Pending;

        public Orders? Order { get; set; }
    }
}

public enum PayMethod
{
    Cash,
    Card,
    UPI,
    Wallet
}

public enum PaymentStatus
{
    Pending,
    Completed,
    Failed
}

File: BookMyTable_Frontend/src/Types/Manager/Payment.ts
export interface Payment {
  paymentId: number;
  orderId: number;
  amount: number;
  paymentDate: string;
  payMethod: PayMethod;
  status: PaymentStatus;
}

export const PayMethod = {
  Cash: 0,
  UPI: 1,
  Card: 2
} as const;

export type PayMethod = typeof PayMethod[keyof typeof PayMethod];

export const PaymentStatus = {
  Pending: 0,
  Completed: 1,
  Failed: 2,
  Refunded: 3
} as const;

export type PaymentStatus = typeof PaymentStatus[keyof typeof PaymentStatus];

export interface PaymentSummary {
  totalPayments: number;
  todayPayments: number;
  completedPayments: number;
  pendingPayments: number;
  totalAmount: number;
  todayAmount: number;
  cashPayments: number;
  upiPayments: number;
}